aws/s3_test.go

```diff
@@ -132,332 +163,277 @@ func (b TestS3Bucket) addObject(s SetupInfo, fileName string, fileBody string) e
+		}
 
-	targetRegions := []string{s.region}
-
-	// Verify that - before creating bucket - it should not exist
-	//
-	// Please note that we are not reusing s.awsSession and creating a random session in a region other
-	// than the one in which the bucket is created - this is useful to test the sceanrio where the user has
-	// AWS_DEFAULT_REGION set to region x but the bucket is in region y.
-	bucketNamesPerRegion, err := getAllS3Buckets(awsSession, time.Now().Add(1*time.Hour*-1), targetRegions, s.bucketName, batchSize)
-	if batchSize < 0 {
-		if err == nil {
-			assert.Fail(t, "Did not fail for invalid batch size")
+		if err != nil {
+			assert.Failf(t, "Failed to list S3 Buckets", errors.WithStackTrace(err).Error())
 		}
-		return
-	}
 
-	if err != nil {
-		assert.Failf(t, "Failed to list S3 Buckets", errors.WithStackTrace(err).Error())
-	} else {
-		assert.NotContains(t, bucketNamesPerRegion[s.region], s.bucketName)
-	}
+		// Validate test bucket does not exist before creation
+		assert.NotContains(t, bucketNamesPerRegion[awsParams.region], aws.String(bucket.name))
 
-	// Create test bucket
-	bucket := TestS3Bucket{
-		name: s.bucketName,
-	}
-	if bucketTags != nil && len(bucketTags) > 0 {
-		bucket.tags = bucketTags
-	}
-	err = bucket.create(s.svc)
-	if err != nil {
-		assert.Failf(t, "Failed to create test bucket", errors.WithStackTrace(err).Error())
-	}
+		// Create test bucket
+		if args.bucketTags != nil && len(args.bucketTags) > 0 {
+			bucket.tags = args.bucketTags
+		}
+		err = bucket.create(awsParams.svc)
+		if err != nil {
+			assert.Failf(t, "Failed to create test bucket", errors.WithStackTrace(err).Error())
+		}
 
-	bucketNamesPerRegion, err = getAllS3Buckets(awsSession, time.Now().Add(1*time.Hour), targetRegions, s.bucketName, batchSize)
-	if err != nil {
-		assert.Failf(t, "Failed to list S3 Buckets", errors.WithStackTrace(err).Error())
-	}
+		bucketNamesPerRegion, err = getAllS3Buckets(awsSession, time.Now().Add(1*time.Hour), targetRegions, bucket.name, args.batchSize)
+		if err != nil {
+			assert.Failf(t, "Failed to list S3 Buckets", errors.WithStackTrace(err).Error())
+		}
 
-	if bucketTags != nil && len(bucketTags) > 0 {
-		for _, tag := range bucketTags {
-			key := strings.ToLower(tag["Key"])
-			value := strings.ToLower(tag["Value"])
-			if key == "cloud-nuke-excluded" && value == "true" {
-				assert.NotContains(t, bucketNamesPerRegion[s.region], aws.String(s.bucketName))
-			}
+		if args.shouldMatch {
+			assert.Contains(t, bucketNamesPerRegion[awsParams.region], aws.String(bucket.name))
+			t.Logf("SUCCESS: Matched bucket - %s", bucket.name)
+		} else {
+			assert.NotContains(t, bucketNamesPerRegion[awsParams.region], aws.String(bucket.name))
+			t.Logf("SUCCESS: Did not match bucket - %s", bucket.name)
 		}
-	}
+	}, nil
+}
 
-	if shouldMatch {
-		assert.Contains(t, bucketNamesPerRegion[s.region], aws.String(s.bucketName))
+// TestListS3Bucket tests listing S3 bucket operation
+func TestListS3Bucket(t *testing.T) {
```

Comment:
`t.Parallel()`? Each test should have this, unless we have a good reason not to run it in parallel?

Reply:
Done.